Fix checkfaillog.c

```shell
gcc -g   -c -o checkfaillog.o checkfaillog.c
checkfaillog.c: In function ‘main’:
checkfaillog.c:48:16: error: implicit declaration of function ‘getuid’; did you mean ‘getutid’? [-Wimplicit-function-declaration]
   48 |     uid_t uid= getuid();
      |                ^~~~~~
      |                getutid
checkfaillog.c:79:13: error: implicit declaration of function ‘exit’ [-Wimplicit-function-declaration]
   79 |             exit(2);
      |             ^~~~
checkfaillog.c:41:1: note: include ‘<stdlib.h>’ or provide a declaration of ‘exit’
   40 | #include "fail_log.h"
  +++ |+#include <stdlib.h>
   41 | 
checkfaillog.c:79:13: warning: incompatible implicit declaration of built-in function ‘exit’ [-Wbuiltin-declaration-mismatch]
   79 |             exit(2);
      |             ^~~~
checkfaillog.c:79:13: note: include ‘<stdlib.h>’ or provide a declaration of ‘exit’
checkfaillog.c:86:13: warning: incompatible implicit declaration of built-in function ‘exit’ [-Wbuiltin-declaration-mismatch]
   86 |             exit(3);
      |             ^~~~
checkfaillog.c:86:13: note: include ‘<stdlib.h>’ or provide a declaration of ‘exit’
checkfaillog.c:103:5: warning: incompatible implicit declaration of built-in function ‘exit’ [-Wbuiltin-declaration-mismatch]
  103 |     exit(0);
      |     ^~~~
checkfaillog.c:103:5: note: include ‘<stdlib.h>’ or provide a declaration of ‘exit’
make: *** [<builtin>: checkfaillog.o] Error 1
```

--- a/pwauth/Makefile
+++ b/pwauth/Makefile
@@ -8,7 +8,7 @@
 # Settings for author's system (Redhat 6.1)
 CC=gcc
 LIB= -lcrypt
-LOCALFLAGS= -g 
+LOCALFLAGS= -g
 
 # For PAM on Redhat Linux
 # LIB=-lpam -ldl

--- a/pwauth/auth_pam.c
+++ b/pwauth/auth_pam.c
@@ -6,7 +6,7 @@
  * are met:
  *
  * 1. Redistributions of source code must retain the above copyright
- *    notice, this list of conditions and the following disclaimer. 
+ *    notice, this list of conditions and the following disclaimer.
  *
  * 2. Redistributions in binary form must reproduce the above copyright
  *    notice, this list of conditions and the following disclaimer in
@@ -56,9 +56,9 @@
 /* Application data structure passed to PAM_conv: */
 
 struct ad_user {
-	char *login;
-	char *passwd;
-	};
+  char *login;
+  char *passwd;
+};
 
 /* The pam_unix.so library in Solaris 2.6 fails to pass along appdata_ptr
  * when it calls the conversation function.  So we use a global variable.  */
@@ -76,121 +76,118 @@ struct ad_user user_info;
 /* In Solaris PAM, msg is a pointer to a pointer to a size num_msg array of
  * pam_message structures.
  */
-# define msgi(i) ((*msg)[i])
+#define msgi(i) ((*msg)[i])
 #else
 /* In Linux PAM and OpenPAM, msg is a pointer to a size num_msg array of
- * pointers to pam_message structures. 
+ * pointers to pam_message structures.
  */
-# define msgi(i) (*(msg[i]))
+#define msgi(i) (*(msg[i]))
 #endif
 
 #ifdef PAM_SOLARIS_26
-int PAM_conv (int num_msg, struct pam_message **msg,
-     struct pam_response **resp, void *appdata_ptr)
-{
-    struct ad_user *user= &user_info;
+int PAM_conv(int num_msg, struct pam_message **msg, struct pam_response **resp,
+             void *appdata_ptr) {
+  struct ad_user *user = &user_info;
 #else
-int PAM_conv (int num_msg, const struct pam_message **msg,
-     struct pam_response **resp, void *appdata_ptr)
-{
-    struct ad_user *user= (struct ad_user *)appdata_ptr;
+int PAM_conv(int num_msg, const struct pam_message **msg,
+             struct pam_response **resp, void *appdata_ptr) {
+  struct ad_user *user = (struct ad_user *)appdata_ptr;
 #endif /* PAM_SOLARIS_26 */
-    struct pam_response *response;
-    int i;
+  struct pam_response *response;
+  int i;
 
-    /* Sanity checking */
-    if (msg == NULL || resp == NULL || user == NULL)
-    	return PAM_CONV_ERR;
+  /* Sanity checking */
+  if (msg == NULL || resp == NULL || user == NULL)
+    return PAM_CONV_ERR;
 
 #ifdef PAM_SOLARIS
-    if (*msg == NULL)
-	return PAM_CONV_ERR;
+  if (*msg == NULL)
+    return PAM_CONV_ERR;
 #endif
 
-    response= (struct pam_response *)
-    	malloc(num_msg * sizeof(struct pam_response));
+  response =
+      (struct pam_response *)malloc(num_msg * sizeof(struct pam_response));
 
-    for (i= 0; i < num_msg; i++)
-    {
-	response[i].resp_retcode= 0;
-	response[i].resp= NULL;
+  for (i = 0; i < num_msg; i++) {
+    response[i].resp_retcode = 0;
+    response[i].resp = NULL;
 
-	switch (msgi(i).msg_style)
-	{
-	case PAM_PROMPT_ECHO_ON:
-	    /* Store the login as the response */
-	    /* This likely never gets called, since login was on pam_start() */
-	    response[i].resp= appdata_ptr ? (char *)strdup(user->login) : NULL;
-	    break;
+    switch (msgi(i).msg_style) {
+    case PAM_PROMPT_ECHO_ON:
+      /* Store the login as the response */
+      /* This likely never gets called, since login was on pam_start() */
+      response[i].resp = appdata_ptr ? (char *)strdup(user->login) : NULL;
+      break;
 
-	case PAM_PROMPT_ECHO_OFF:
-	    /* Store the password as the response */
-	    response[i].resp= appdata_ptr ? (char *)strdup(user->passwd) : NULL;
-	    break;
+    case PAM_PROMPT_ECHO_OFF:
+      /* Store the password as the response */
+      response[i].resp = appdata_ptr ? (char *)strdup(user->passwd) : NULL;
+      break;
 
-	case PAM_TEXT_INFO:
-	case PAM_ERROR_MSG:
-	    /* Shouldn't happen since we have PAM_SILENT set. If it happens
-	     * anyway, ignore it. */
-	    break;
+    case PAM_TEXT_INFO:
+    case PAM_ERROR_MSG:
+      /* Shouldn't happen since we have PAM_SILENT set. If it happens
+       * anyway, ignore it. */
+      break;
 
-	default:
-	    /* Something strange... */
-	    if (response != NULL) free(response);
-	    return PAM_CONV_ERR;
-	}
+    default:
+      /* Something strange... */
+      if (response != NULL)
+        free(response);
+      return PAM_CONV_ERR;
     }
-    /* On success, return the response structure */
-    *resp= response;
-    return PAM_SUCCESS;
+  }
+  /* On success, return the response structure */
+  *resp = response;
+  return PAM_SUCCESS;
 }
-
 
 /* CHECK_AUTH - Check a login and return a status code.
  * (This version for systems using PAM.)
  */
 
-int check_auth(char *login, char *passwd)
-{
+int check_auth(char *login, char *passwd) {
 #ifndef PAM_SOLARIS_26
-    struct ad_user user_info= {login, passwd};
+  struct ad_user user_info = {login, passwd};
 #endif /* PAM_SOLARIS_26 */
-    struct pam_conv conv= { PAM_conv, (void *)&user_info };
-    pam_handle_t *pamh= NULL;
-    int retval;
+  struct pam_conv conv = {PAM_conv, (void *)&user_info};
+  pam_handle_t *pamh = NULL;
+  int retval;
 
 #ifdef NEED_UID
-    struct passwd *pwd;
+  struct passwd *pwd;
 
-    if ((pwd= getpwnam(login)) == NULL) return STATUS_UNKNOWN;
-    hisuid= pwd->pw_uid;
-    haveuid= 1;
+  if ((pwd = getpwnam(login)) == NULL)
+    return STATUS_UNKNOWN;
+  hisuid = pwd->pw_uid;
+  haveuid = 1;
 #endif
 #ifdef MIN_UNIX_UID
-    if (hisuid < MIN_UNIX_UID) return STATUS_BLOCKED;
+  if (hisuid < MIN_UNIX_UID)
+    return STATUS_BLOCKED;
 #endif
 
 #ifdef PAM_SOLARIS_26
-    user_info.login= login;
-    user_info.passwd= passwd;
+  user_info.login = login;
+  user_info.passwd = passwd;
 #endif /* PAM_SOLARIS_26 */
 
-    retval= pam_start("pwauth", login, &conv, &pamh);
+  retval = pam_start("pwauth", login, &conv, &pamh);
 
-    if (retval == PAM_SUCCESS)
-	retval= pam_authenticate(pamh, PAM_SILENT);
+  if (retval == PAM_SUCCESS)
+    retval = pam_authenticate(pamh, PAM_SILENT);
 
-    if (retval == PAM_SUCCESS)
-	retval= pam_acct_mgmt(pamh, 0);		/* permitted access? */
+  if (retval == PAM_SUCCESS)
+    retval = pam_acct_mgmt(pamh, 0); /* permitted access? */
 
-    if (pam_end(pamh,retval) != PAM_SUCCESS)	/* close PAM */
-    {
-	pamh= NULL;
-	return STATUS_INT_ERR;
-    }
+  if (pam_end(pamh, retval) != PAM_SUCCESS) /* close PAM */
+  {
+    pamh = NULL;
+    return STATUS_INT_ERR;
+  }
 
-    /* On failure we always return STATUS_UNKNOWN although we can't tell
-     * if the failure was because of a bad login or a bad password */
-    return (retval == PAM_SUCCESS ? STATUS_OK : STATUS_UNKNOWN);
+  /* On failure we always return STATUS_UNKNOWN although we can't tell
+   * if the failure was because of a bad login or a bad password */
+  return (retval == PAM_SUCCESS ? STATUS_OK : STATUS_UNKNOWN);
 }
-#endif  /* PAM */
+#endif /* PAM */

--- a/pwauth/auth_sun.c
+++ b/pwauth/auth_sun.c
@@ -6,7 +6,7 @@
  * are met:
  *
  * 1. Redistributions of source code must retain the above copyright
- *    notice, this list of conditions and the following disclaimer. 
+ *    notice, this list of conditions and the following disclaimer.
  *
  * 2. Redistributions in binary form must reproduce the above copyright
  *    notice, this list of conditions and the following disclaimer in
@@ -36,17 +36,18 @@
 #ifdef SHADOW_SUN
 #ifdef NEED_UID
 #include <pwd.h>
+#include <time.h>
 #endif
 #include <shadow.h>
-struct spwd *getspnam();
-char *crypt();
+// struct spwd *getspnam();
+// char *crypt();
 #endif /* SHADOW_SUN */
 
 #ifdef SHADOW_JFH
 #ifdef NEED_UID
 #include <pwd.h>
 #endif
-#include <shadow.h>	  /* this may be hidden in /usr/local/include */
+#include <shadow.h> /* this may be hidden in /usr/local/include */
 struct spwd *getspnam();
 char *pw_encrypt();
 #endif /* SHADOW_JFH */
@@ -54,56 +55,56 @@ char *pw_encrypt();
 #if defined(SHADOW_JFH) || defined(SHADOW_SUN)
 /* ===================== JFH and SUN Authentication ===================== */
 
-
 /* CHECK_AUTH - Check a login and return a status code.
  * (This version for systems with getspnam() call.)
  */
 
-int check_auth(char *login, char *passwd)
-{
-    char *cpass;
-    struct spwd *spwd= getspnam(login);
+int check_auth(char *login, char *passwd) {
+  char *cpass;
+  struct spwd *spwd = getspnam(login);
 #ifdef NEED_UID
-    struct passwd *pwd;
+  struct passwd *pwd;
 #endif
 #if defined(CHECK_LOGIN_EXPIRATION) || defined(CHECK_PASSWORD_EXPIRATION)
-    int today= time(NULL)/(24*60*60);
+  int today = time(NULL) / (24 * 60 * 60);
 #endif
-    if (spwd == NULL)
-    	return(errno == EACCES ? STATUS_INT_NOROOT : STATUS_UNKNOWN);
+  if (spwd == NULL)
+    return (errno == EACCES ? STATUS_INT_NOROOT : STATUS_UNKNOWN);
 #ifdef NEED_UID
-    if ((pwd= getpwnam(login)) == NULL) return(STATUS_UNKNOWN);
-    hisuid= pwd->pw_uid;
-    haveuid= 1;
+  if ((pwd = getpwnam(login)) == NULL)
+    return (STATUS_UNKNOWN);
+  hisuid = pwd->pw_uid;
+  haveuid = 1;
 #endif
 #ifdef MIN_UNIX_UID
-    if (hisuid < MIN_UNIX_UID) return(STATUS_BLOCKED);
+  if (hisuid < MIN_UNIX_UID)
+    return (STATUS_BLOCKED);
 #endif
 #ifdef SHADOW_JFH
-    cpass= pw_encrypt(passwd, spwd->sp_pwdp);
+  cpass = pw_encrypt(passwd, spwd->sp_pwdp);
 #else
-    cpass= crypt(passwd, spwd->sp_pwdp);
+  cpass = crypt(passwd, spwd->sp_pwdp);
 #endif
-    if (!cpass || strcmp(cpass, spwd->sp_pwdp)) return STATUS_INVALID;
+  if (!cpass || strcmp(cpass, spwd->sp_pwdp))
+    return STATUS_INVALID;
 #ifdef CHECK_LOGIN_EXPIRATION
-    if (spwd->sp_expire >= 0 && spwd->sp_expire < today)
-	return STATUS_EXPIRED;
+  if (spwd->sp_expire >= 0 && spwd->sp_expire < today)
+    return STATUS_EXPIRED;
 #endif
 #ifdef CHECK_PASSWORD_EXPIRATION
-    /* Root forced password expiration */
-    if (spwd->sp_lstchg == 0)
-	return STATUS_PW_EXPIRED;
-	    
-    /* Normal password expiration */
-    /* We used to have sp_lstchg + sp_max + sp_inact < today
-     * here, apparantly during the inact period you are only supposed to be able
-     * to get on to change your password, not anything else, show we shouldn't
-     * allow access during that period.
-     */
-    if (spwd->sp_max >= 0 &&
-        spwd->sp_lstchg + spwd->sp_max < today)
-	return STATUS_PW_EXPIRED;
+  /* Root forced password expiration */
+  if (spwd->sp_lstchg == 0)
+    return STATUS_PW_EXPIRED;
+
+  /* Normal password expiration */
+  /* We used to have sp_lstchg + sp_max + sp_inact < today
+   * here, apparantly during the inact period you are only supposed to be able
+   * to get on to change your password, not anything else, show we shouldn't
+   * allow access during that period.
+   */
+  if (spwd->sp_max >= 0 && spwd->sp_lstchg + spwd->sp_max < today)
+    return STATUS_PW_EXPIRED;
 #endif
-    return STATUS_OK;
+  return STATUS_OK;
 }
 #endif /* SHADOW_JFH || SHADOW_SUN */

--- a/pwauth/checkfaillog.c
+++ b/pwauth/checkfaillog.c
@@ -6,7 +6,7 @@
  * are met:
  *
  * 1. Redistributions of source code must retain the above copyright
- *    notice, this list of conditions and the following disclaimer. 
+ *    notice, this list of conditions and the following disclaimer.
  *
  * 2. Redistributions in binary form must reproduce the above copyright
  *    notice, this list of conditions and the following disclaimer in
@@ -31,77 +31,75 @@
  * =======================================================================
  */
 
-#include <stdio.h>
-#include <sys/types.h>
-#include <utmp.h>
 #include <pwd.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <utmp.h>
 
 #include "config.h"
 #include "fail_log.h"
 
+int main(int argc, char **argv) {
+  int i, j;
+  int reset = 0, verbose = 1;
+  char *user = NULL, *msg = NULL;
+  uid_t uid = getuid();
 
-int main(int argc, char **argv)
-{
-    int i, j;
-    int reset= 0, verbose= 1;
-    char *user= NULL, *msg= NULL;
-    uid_t uid= getuid();
+  /* Parse command line */
+  for (i = 1; i < argc; i++) {
+    if (argv[i][0] == '-') {
+      for (j = 1; argv[i][j] != '\0'; j++) {
+        switch (argv[i][j]) {
+        case 'z':
+          reset = 1;
+          break;
+        case 's':
+          verbose = 0;
+          break;
+        default:
+          goto usage;
+        }
+      }
+    } else if (user == NULL)
+      user = argv[i];
+    else
+      goto usage;
+  }
 
-    /* Parse command line */
-    for (i= 1; i < argc; i++)
-    {
-	if (argv[i][0] == '-')
-	{
-	    for (j= 1; argv[i][j] != '\0'; j++)
-	    {
-		switch (argv[i][j])
-		{
-		    case 'z': reset= 1; break;
-		    case 's': verbose= 0; break;
-		    default: goto usage;
-		}
-	    }
-	}
-	else if (user == NULL)
-	    user= argv[i];
-	else
-	    goto usage;
+  /* Root can run this on other user's accounts */
+  if (user != NULL) {
+    struct passwd *pw;
+
+    if ((pw = getpwnam(user)) == NULL) {
+      fprintf(stderr, "%s: User %s does not exist.\n", argv[0], user);
+      exit(2);
     }
 
-    /* Root can run this on other user's accounts */
-    if (user != NULL)
-    {
-	struct passwd *pw;
-
-	if ((pw= getpwnam(user)) == NULL)
-	{
-	    fprintf(stderr,"%s: User %s does not exist.\n", argv[0], user);
-	    exit(2);
-	}
-
-	if (uid != 0 && pw->pw_uid != uid)
-	{
-	    fprintf(stderr,"%s: Only root can access other user's accounts.\n",
-		    argv[0]);
-	    exit(3);
-	}
-
-	uid= pw->pw_uid;
+    if (uid != 0 && pw->pw_uid != uid) {
+      fprintf(stderr, "%s: Only root can access other user's accounts.\n",
+              argv[0]);
+      exit(3);
     }
 
-    /* Do the thing */
+    uid = pw->pw_uid;
+  }
+
+  /* Do the thing */
 #if defined(FAILLOG_JFH) || defined(FAILLOG_OPENBSD)
-    msg= check_fails(uid, reset, verbose);
+  msg = check_fails(uid, reset, verbose);
 #endif
 
-    /* Output the result */
-    if (msg != NULL)
-    	puts(msg);
-    else if (!verbose)
-    	puts("0:::");
+  /* Output the result */
+  if (msg != NULL)
+    puts(msg);
+  else if (!verbose)
+    puts("0:::");
 
-    exit(0);
+  exit(0);
 
-usage: fprintf(stderr,"usage: %s [-z] [-s] [user]\n", argv[0]);
-    exit(1);
+usage:
+  fprintf(stderr, "usage: %s [-z] [-s] [user]\n", argv[0]);
+  exit(1);
 }

--- a/pwauth/config.h
+++ b/pwauth/config.h
@@ -119,22 +119,22 @@
 
 /* LOW-LEVEL OPTIONS */
 
-/* #define SHADOW_NONE		/**/
-/* #define SHADOW_BSD		/* FreeBSD, NetBSD, OpenBSD, BSDI, OS X */
-#define SHADOW_SUN /* Linux, Solaris, IRIX */
-/* #define SHADOW_JFH		/**/
-/* #define SHADOW_MDW		/**/
-/* #define SHADOW_AIX		/* AIX (see also AUTHENTICATE_AIX) */
-/* #define SHADOW_HPUX		/* HPUX ? */
+// #define SHADOW_NONE		/**/
+// #define SHADOW_BSD		/* FreeBSD, NetBSD, OpenBSD, BSDI, OS X */
+// #define SHADOW_SUN /* Linux, Solaris, IRIX */
+// #define SHADOW_JFH		/**/
+// #define SHADOW_MDW		/**/
+// #define SHADOW_AIX		/* AIX (see also AUTHENTICATE_AIX) */
+// #define SHADOW_HPUX		/* HPUX ? */
 
 /* HIGH-LEVEL OPTIONS */
 
-/* #define PAM			/* Linux PAM or OpenPAM */
-/* #define PAM_OLD_OS_X		/* PAM on OS X version 10.5 or older */
-/* #define PAM_SOLARIS		/* PAM on Solaris other than 2.6 */
-/* #define PAM_SOLARIS_26	/* PAM on Solaris 2.6 */
-/* #define LOGIN_CONF_OPENBSD	/* login.conf on OpenBSD */
-/* #define AUTHENTICATE_AIX	/* AIX authenticate() function */
+// #define PAM /* Linux PAM or OpenPAM */
+// #define PAM_OLD_OS_X		/* PAM on OS X version 10.5 or older */
+// #define PAM_SOLARIS		/* PAM on Solaris other than 2.6 */
+// #define PAM_SOLARIS_26	/* PAM on Solaris 2.6 */
+// #define LOGIN_CONF_OPENBSD	/* login.conf on OpenBSD */
+// #define AUTHENTICATE_AIX	/* AIX authenticate() function */
 
 /* There is also limited support for three failure logging systems (the
  * database that informs you that "there have been 3426 unsuccessful attempts
@@ -222,8 +222,8 @@
  * right with PAM.
  */
 
-#define NOLOGIN_FILE "/etc/nologin" /**/
-#define MIN_NOLOGIN_UID 1           /**/
+// #define NOLOGIN_FILE "/etc/nologin" /**/
+// #define MIN_NOLOGIN_UID 1           /**/
 
 /* Defining CHECK_LOGIN_EXPIRATION and CHECK_PASSWORD_EXPIRATION causes
  * pwauth to check if the given login has expired, or it's password has

--- a/pwauth/main.c
+++ b/pwauth/main.c
@@ -117,7 +117,7 @@ int main(int argc, char **argv) {
 #endif
 
   if (status == STATUS_OK) {
-    /* Good login */
+/* Good login */
 #ifdef UNIX_LASTLOG
     lastlog();
 #endif

--- a/pwauth/nologin.c
+++ b/pwauth/nologin.c
@@ -6,7 +6,7 @@
  * are met:
  *
  * 1. Redistributions of source code must retain the above copyright
- *    notice, this list of conditions and the following disclaimer. 
+ *    notice, this list of conditions and the following disclaimer.
  *
  * 2. Redistributions in binary form must reproduce the above copyright
  *    notice, this list of conditions and the following disclaimer in
@@ -38,16 +38,15 @@
  * uid is below MIN_NOLOGIN_UID.
  */
 
-check_nologin()
-{
-    /* Return true if the file does not exist (the access() function returns
-     * true if the file doesn't exist - pretty dumb, eh?) */
-    if (access(NOLOGIN_FILE, F_OK))
-	return 1;
+int check_nologin() {
+  /* Return true if the file does not exist (the access() function returns
+   * true if the file doesn't exist - pretty dumb, eh?) */
+  if (access(NOLOGIN_FILE, F_OK))
+    return 1;
 #if MIN_NOLOGIN_UID > 0
-    if (hisuid < MIN_NOLOGIN_UID)
-	return 1;
+  if (hisuid < MIN_NOLOGIN_UID)
+    return 1;
 #endif
-    return 0;
+  return 0;
 }
 #endif

--- a/pwauth/pwauth.h
+++ b/pwauth/pwauth.h
@@ -6,7 +6,7 @@
  * are met:
  *
  * 1. Redistributions of source code must retain the above copyright
- *    notice, this list of conditions and the following disclaimer. 
+ *    notice, this list of conditions and the following disclaimer.
  *
  * 2. Redistributions in binary form must reproduce the above copyright
  *    notice, this list of conditions and the following disclaimer in
@@ -31,17 +31,17 @@
  * =======================================================================
  */
 
+#include <errno.h>
+#include <fcntl.h>
 #include <stdio.h>
 #include <stdlib.h>
-#include <unistd.h>
 #include <string.h>
-#include <fcntl.h>
-#include <errno.h>
-#include <sys/types.h>
-#include <sys/stat.h>
-#include <sys/time.h>
 #include <sys/file.h>
 #include <sys/resource.h>
+#include <sys/stat.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <unistd.h>
 /* Configuration is meant to be done in config.h */
 #include "config.h"
 
@@ -52,67 +52,66 @@
  * returned for both.
  */
 
-#define STATUS_OK 	  0   /* Valid Login */
-#define STATUS_UNKNOWN	  1   /* Login doesn't exist or password incorrect */
-#define STATUS_INVALID	  2   /* Password was incorrect */
-#define STATUS_BLOCKED    3   /* UID is below minimum allowed to use this */
-#define STATUS_EXPIRED    4   /* Login ID has passed it's expiration date */
-#define STATUS_PW_EXPIRED 5   /* Password has expired and must be changed */
-#define STATUS_NOLOGIN    6   /* Logins have been turned off */
-#define STATUS_MANYFAILS  7   /* Bad login limit exceeded */
+#define STATUS_OK 0         /* Valid Login */
+#define STATUS_UNKNOWN 1    /* Login doesn't exist or password incorrect */
+#define STATUS_INVALID 2    /* Password was incorrect */
+#define STATUS_BLOCKED 3    /* UID is below minimum allowed to use this */
+#define STATUS_EXPIRED 4    /* Login ID has passed it's expiration date */
+#define STATUS_PW_EXPIRED 5 /* Password has expired and must be changed */
+#define STATUS_NOLOGIN 6    /* Logins have been turned off */
+#define STATUS_MANYFAILS 7  /* Bad login limit exceeded */
 
-#define STATUS_INT_USER  50   /* pwauth was run by wrong uid */
-#define STATUS_INT_ARGS  51   /* login/password not passed in correctly */
-#define STATUS_INT_ERR   52   /* Miscellaneous internal errors */
-#define STATUS_INT_NOROOT 53  /* pwauth cannot read password database */
-
+#define STATUS_INT_USER 50   /* pwauth was run by wrong uid */
+#define STATUS_INT_ARGS 51   /* login/password not passed in correctly */
+#define STATUS_INT_ERR 52    /* Miscellaneous internal errors */
+#define STATUS_INT_NOROOT 53 /* pwauth cannot read password database */
 
 #ifdef IGNORE_CASE
-# include <ctype.h>
+#include <ctype.h>
 #endif
 
 #ifdef UNIX_LASTLOG
-# define NEED_UID
-# include <utmp.h>
-# ifdef HAVE_LASTLOG_H
-#  include <lastlog.h>
-# endif
-# ifndef UT_LINESIZE
-#  define UT_LINESIZE 8
-# endif
-# ifndef UT_HOSTSIZE
-#  define UT_HOSTSIZE 16
-# endif
-# if !defined(LASTLOG) && defined(PATHS_H)
-#  include <paths.h>
-#  ifdef _PATH_LASTLOG
-#   define LASTLOG _PATH_LASTLOG
-#  endif
-# endif
-# ifndef LASTLOG
-#  define LASTLOG "/usr/adm/lastlog"
-# endif
+#define NEED_UID
+#include <utmp.h>
+#ifdef HAVE_LASTLOG_H
+#include <lastlog.h>
+#endif
+#ifndef UT_LINESIZE
+#define UT_LINESIZE 8
+#endif
+#ifndef UT_HOSTSIZE
+#define UT_HOSTSIZE 16
+#endif
+#if !defined(LASTLOG) && defined(PATHS_H)
+#include <paths.h>
+#ifdef _PATH_LASTLOG
+#define LASTLOG _PATH_LASTLOG
+#endif
+#endif
+#ifndef LASTLOG
+#define LASTLOG "/usr/adm/lastlog"
+#endif
 #endif /*UNIX_LASTLONG*/
 
 #ifdef NOLOGIN_FILE
-# ifndef MIN_NOLOGIN_UID
-#  define MIN_NOLOGIN_UID 0
-# endif
-# if MIN_NOLOGIN_UID > 0
-#  define NEED_UID
-# endif
+#ifndef MIN_NOLOGIN_UID
+#define MIN_NOLOGIN_UID 0
+#endif
+#if MIN_NOLOGIN_UID > 0
+#define NEED_UID
+#endif
 #endif
 
 #include "fail_log.h"
 
 #ifdef MIN_UNIX_UID
-# if MIN_UNIX_UID <= 0
-#  undef MIN_UNIX_UID
-# else
-#  ifndef NEED_UID
-#   define NEED_UID
-#  endif
-# endif
+#if MIN_UNIX_UID <= 0
+#undef MIN_UNIX_UID
+#else
+#ifndef NEED_UID
+#define NEED_UID
+#endif
+#endif
 #endif
 
 #ifdef NEED_UID
@@ -121,11 +120,11 @@ extern int haveuid;
 #endif
 
 #ifndef SLEEP_TIME
-# define SLEEP_TIME 2
+#define SLEEP_TIME 2
 #endif
 
 #ifndef BFSZ
-# define BFSZ 1024
+#define BFSZ 1024
 #endif
 
 void snooze(int seconds);

